### PR TITLE
CASSANDRA-18124 Make keystore_password nullable

### PR DESCRIPTION
**Pull request by maulin-vasavada**
Thanks for sending a pull request! Here are some tips if you're new here:
 
 * Ensure you have added or run the [appropriate tests](https://cassandra.apache.org/_/development/testing.html) for your PR.
 * Be sure to keep the PR description updated to reflect all changes.
 * Write your PR title to summarize what this PR proposes.
 * If possible, provide a concise example to reproduce the issue for a faster review.
 * Read our [contributor guidelines](https://cassandra.apache.org/_/development/index.html)
 * If you're making a documentation change, see our [guide to documentation contribution](https://cassandra.apache.org/_/development/documentation.html)
 
Commit messages should follow the following format:

```
<One sentence description, usually Jira title or CHANGES.txt summary>

<Optional lengthier description (context on patch)>

patch by <Authors>; reviewed by <Reviewers> for CASSANDRA-#####

Co-authored-by: Name1 <email1>
Co-authored-by: Name2 <email2>

```

The [Cassandra Jira](https://issues.apache.org/jira/projects/CASSANDRA/issues/)

### CODE REVIEW

**Comment by rtib** [comment]
LGTM

File: examples/ssl-factory/src/org/apache/cassandra/security/KubernetesSecretsPEMSslContextFactory.java [hunk 1]
```diff
@@ -135,12 +137,12 @@ public KubernetesSecretsPEMSslContextFactory(Map<String, Object> parameters)
         super(parameters);
 
         pemEncodedKey = getValueFromEnv(getString(PEMConfigKey.PRIVATE_KEY_ENV_VAR, DEFAULT_PRIVATE_KEY_ENV_VAR_NAME),
-                                        DEFAULT_PRIVATE_KEY);
```
Comment (smiklosovic) [inline examples/ssl-factory/src/org/apache/cassandra/security/KubernetesSecretsPEMSslContextFactory.java:138]: @maulin-vasavada  there are all these instances of changed formatting, can we return it back as it was? it happens in the below class too

File: examples/ssl-factory/src/org/apache/cassandra/security/KubernetesSecretsPEMSslContextFactory.java [hunk 2]
```diff
@@ -181,4 +183,4 @@ protected TrustManagerFactory buildTrustManagerFactory() throws SSLException
         String PRIVATE_KEY_PASSWORD_ENV_VAR = "PRIVATE_KEY_PASSWORD_ENV_VAR";
         String TRUSTED_CERTIFICATE_ENV_VAR = "TRUSTED_CERTIFICATE_ENV_VAR";
     }
-}
+}
```
Comment (smiklosovic) [inline examples/ssl-factory/src/org/apache/cassandra/security/KubernetesSecretsPEMSslContextFactory.java:186]: can you please check your settings and change this? this happens in every class you touched.

File: examples/ssl-factory/src/org/apache/cassandra/security/KubernetesSecretsPEMSslContextFactory.java (hunk 1 above)
Comment (maulin-vasavada) [reply examples/ssl-factory/src/org/apache/cassandra/security/KubernetesSecretsPEMSslContextFactory.java:138]: ack. I seem to have the settings issue again. Let me see. I did run ant generate-idea-files before I started to make changes.

**Comment by maulin-vasavada** [comment]
Will close this PR and raise another one with formatting fixes.

**Pull request closed without merging**